### PR TITLE
Outsource aggregate translator configuration again

### PR DESCRIPTION
**Pull request by dropdevcoding**
Please have a look at this solution. If this is okay for you I will readd some more unit tests to ensure the static factory method to work properly

### CODE REVIEW

**Comment by codeliner** [comment]
:+1: looks good

**Comment by dropdevcoding** [comment]
I merged the former unit tests for ConfigrableAggregateTranslator when using AggregateTranslatorConfiguration but leaving the old ones for constructor configuration since both cases have to be tested for the BC solution

**Comment by codeliner** [comment]
@dropdevcoding I've updated the docheaders, see https://github.com/prooph/event-store/commit/e1b0246bc65593137cc52b7708e4a18855bde1e0
can you merge latest master into your PR and align docheader of the new file then travis should turn green and we can merge. Thx

**Comment by dropdevcoding** [comment]
Oh, okay, I wanted to do that after the merge in a separate PR since @prolic asked me to.

I will do the merge and fix the headers.

**Comment by codeliner** [comment]
yeah, our travis badge was red on the master branch. So I thought it is a good idea to fix that quickly :D

**Comment by dropdevcoding** [comment]
Done. Are there any docs to be updated?

***Off-Topic***: Ever spent a thought on using phing and having wrapped the scripts defined for travis in one single build task in a build.xml file? I think this would help to contribute clean PRs since you only had to run ```./vendor/bin/phing build``` instead of looking up the scripts and execute them separately prior a push.

**Comment by prolic** [comment]
@dropdevcoding this is already in place within composer.json

about docs, see https://github.com/prooph/event-store/tree/master/docs and provide a little documentation on how to use this new feature.

File: src/Aggregate/AggregateTranslatorConfiguration.php [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+
+    /**
+     * @var null|callable
+     */
+    private $messageToEventCallback;
+
+    /**
+     * AggregateTranslatorConfiguration constructor.
+     *
+     * @param $versionMethodName
```
Comment (prolic) [inline src/Aggregate/AggregateTranslatorConfiguration.php]: `@param string`

File: src/Aggregate/AggregateTranslatorConfiguration.php [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+     * @param $identifierMethodName
```
Comment (prolic) [inline src/Aggregate/AggregateTranslatorConfiguration.php]: `@param string`

File: src/Aggregate/AggregateTranslatorConfiguration.php [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+     * @param $popRecordedEventsMethodName
```
Comment (prolic) [inline src/Aggregate/AggregateTranslatorConfiguration.php]: and so on...

File: src/Aggregate/AggregateTranslatorConfiguration.php [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+        $instance = clone $this;
+        $instance->versionMethodName = $versionMethodName;
+
+        return $instance;
+    }
+
+    /**
+     * Returns a new instance having set the given name of the method which is used to determine the identifier
+     *
+     * @param $identifierMethodName
```
Comment (prolic) [inline src/Aggregate/AggregateTranslatorConfiguration.php]: `@param string`

File: src/Aggregate/AggregateTranslatorConfiguration.php [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+        $instance = clone $this;
+        $instance->identifierMethodName = $identifierMethodName;
+
+        return $instance;
+    }
+
+    /**
+     * Returns a new instance having set the given name of the method which is used to pop recorded events
+     *
+     * @param $popRecordedEventsMethodName
```
Comment (prolic) [inline src/Aggregate/AggregateTranslatorConfiguration.php]: `@param string`

**Comment by prolic** [comment]
:+1:

**Pull request merged**